src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -362,6 +363,13 @@ void CoreSMTSolver::attachClause(CRef cr)
     assert(c.size() > 1);
     watches[~c[0]].push(Watcher(cr, c[1]));
     watches[~c[1]].push(Watcher(cr, c[0]));
+    if(c.size() > 2 )
+        watches[~c[2]].push(Watcher(cr, c[0]));
+    else{
+        next_v[next_v.size() - 1].insert(var(~c[0]));
```

Comment:
Maybe use `next_v.back().insert(var (~c[0]))` here and in similar places.

Reply:
Corrected